pkg/apis/maistra/conversion/extension_providers.go

```diff
@@ -44,31 +44,57 @@ func populateExtensionProvidersValues(in *v2.ControlPlaneSpec, allValues map[str
 			if config.IncludeAdditionalHeadersInCheck != nil {
 				values["includeAdditionalHeadersInCheck"] = mapOfStringToInterface(config.IncludeAdditionalHeadersInCheck)
 			}
-			if config.IncludeRequestBodyInCheck != nil {
-				includeRequestBodyInCheckValues := map[string]interface{}{}
-				if config.IncludeRequestBodyInCheck.MaxRequestBytes != nil {
-					includeRequestBodyInCheckValues["maxRequestBytes"] = *config.IncludeRequestBodyInCheck.MaxRequestBytes
-				}
-				if config.IncludeRequestBodyInCheck.AllowPartialMessage != nil {
-					includeRequestBodyInCheckValues["allowPartialMessage"] = *config.IncludeRequestBodyInCheck.AllowPartialMessage
-				}
-				if config.IncludeRequestBodyInCheck.PackAsBytes != nil {
-					includeRequestBodyInCheckValues["packAsBytes"] = *config.IncludeRequestBodyInCheck.PackAsBytes
-				}
-				values["includeRequestBodyInCheck"] = includeRequestBodyInCheckValues
-			}
+			convertIncludeRequestBodyInCheckConfigToValues(config.IncludeRequestBodyInCheck, values)
 			extensionProvidersValues = append(extensionProvidersValues, map[string]interface{}{
 				"name":              provider.Name,
 				"envoyExtAuthzHttp": values,
 			})
 		}
+		if provider.EnvoyExtAuthzGRPC != nil {
+			config := provider.EnvoyExtAuthzGRPC
+			values := map[string]interface{}{
+				"service": config.Service,
+				"port":    config.Port,
+			}
+			if config.Timeout != nil {
+				values["timeout"] = *config.Timeout
+			}
+			if config.FailOpen != nil {
+				values["failOpen"] = *config.FailOpen
+			}
+			if config.StatusOnError != nil {
+				values["statusOnError"] = *config.StatusOnError
+			}
+
```

Comment:
nit: please remove this line (and optionally add a blank line after lines 69 and 47).